lib/streamlit/runtime/runtime.py

```diff
@@ -150,15 +155,19 @@ def __init__(self, config: RuntimeConfig):
         self._uploaded_file_mgr.on_files_updated.connect(self._on_files_updated)
         self._media_file_mgr = MediaFileManager(storage=config.media_file_storage)
 
+        # Ignore the obviously incorrect type below where we pass None where a
+        # SessionStorage should be in the first argument. This is fine because we're
+        # not yet using the argument in the one class that currently implements
+        # SessionManager.
+        self._session_mgr = config.session_manager_class(None, self)  # type: ignore
```

Comment:
I have a comment below that goes into more detail, but passing `self` to SessionManager feels a bit bad and I wonder if we can change the API to allow SessionManager to be more fully decoupled from Runtime.

Reply:
Hm, yeah. I originally did this to keep the `connect_session` signature from getting too unwieldy, but I think it'd be preferable to pass `SessionData`/ the `UploadedFileManager` / etc into the `SessionManager` directly vs passing the `Runtime` to get access to these things.